migrace/021.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+// Vytvoři tabulyu pro uživatelské slevy
+$this->q("
+CREATE TABLE `slevy` (
+  `id` int NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `id_uzivatele` int(11) NOT NULL,
+  `castka` float NOT NULL,
+  `rok` int NOT NULL,
+  `provedeno` datetime NOT NULL,
+  `provedl` int(11) NULL,
+  `poznamka` text NOT NULL,
+  FOREIGN KEY (`id_uzivatele`) REFERENCES `uzivatele_hodnoty` (`id_uzivatele`) ON DELETE CASCADE,
```

Comment:
U cizích klíču pls na vše restrict.

Reply:
OK, jen bych se tedy zeptal proč?

Reply:
J, primární motivace je, aby se omylem (v kódu nebo při ruční editaci DB) nesmazalo víc věcí naráz, resp. abys musel explicitně říct, že chceš mazat i platby.

Typický use-case je, že někdo vytvořil v DB testovacího / nesmyslného uživatele, Shako či Maník ho jdou přes db smazat a místo 2172 budou mazat 2712, ale existence nějakých plateb ti řekne, že asi mažeš reálného člověka (příp. i chceš smazat reálný účet, ale neuvědomíš si, že takový účet s námi má nějaké finanční závazky, a když to na tebe vyskočí, tak záměr smazat přehodnotíš).